packages/workflow-core/src/lib/utils/context-transformers/helpers-transformer.ts

```diff
@@ -0,0 +1,70 @@
+import { BaseContextTransformer, THelperFormatingLogic } from './types';
+import { TContext } from '../types';
+
+export type THelperMethod = 'regex' | 'imageUrlToBase64';
+export class HelpersTransformer extends BaseContextTransformer {
+  name = 'helpers-transformer';
+  mapping: THelperFormatingLogic;
+
+  constructor(mapping: THelperFormatingLogic) {
+    super();
+    this.mapping = mapping;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async transform(context: TContext, _options = {}) {
+    for (const mappingLogic of this.mapping) {
+      const sourcePath = mappingLogic.source.split('.');
+      const targetPath = mappingLogic.target.split('.');
+
+      const sourceAttributeValue = this.getNestedProperty(context, sourcePath);
+      const transformedValue = await this[mappingLogic.method](
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore - sourceAttributeValue is dynamic and configurable
+        sourceAttributeValue,
+        mappingLogic.value,
+      );
+
+      this.setNestedProperty(context, targetPath, transformedValue);
+    }
+
+    return context;
+  }
+
+  async regex(attribute: string, value: string) {
+    const result = attribute.match(new RegExp(value));
+
+    return Promise.resolve(result ? result[0] : null);
+  }
+
+  async imageUrlToBase64(url: string, _value: string) {
+    const response = await fetch(url);
```

Comment:
*non-blocker - let's move a function to `@ballerine/common` for converting to and from base64.